Exercise_5.java

```diff
@@ -1,36 +1,57 @@
-class IterativeQuickSort { 
-    void swap(int arr[], int i, int j) 
-    { 
-	//Try swapping without extra variable 
-    } 
-  
-    /* This function is same in both iterative and 
-       recursive*/
-    int partition(int arr[], int l, int h) 
-    { 
-        //Compare elements and swap.
-    } 
-  
-    // Sorts arr[l..h] using iterative QuickSort 
-    void QuickSort(int arr[], int l, int h) 
-    { 
-        //Try using Stack Data Structure to remove recursion.
-    } 
-  
-    // A utility function to print contents of arr 
-    void printArr(int arr[], int n) 
-    { 
-        int i; 
-        for (i = 0; i < n; ++i) 
-            System.out.print(arr[i] + " "); 
-    } 
-  
-    // Driver code to test above 
-    public static void main(String args[]) 
-    { 
-        IterativeQuickSort ob = new IterativeQuickSort(); 
-        int arr[] = { 4, 3, 5, 2, 1, 3, 2, 3 }; 
-        ob.QuickSort(arr, 0, arr.length - 1); 
-        ob.printArr(arr, arr.length); 
-    } 
-} 
\ No newline at end of file
+//time complexity - O(8log(8))
+//space complexity - O(8)
+
+
+class IterativeQuickSort {
+    void swap(int arr[], int i, int j) {
+        // Try swapping without extra variable
+    }
+
+    /*
+     * This function is same in both iterative and
+     * recursive
+     */
+    int partition(int arr[], int l, int h) {
+        // Compare elements and swap.
+        int pivot = arr[h];
+        int i = (l - 1);
+        for (int j = l; j <= h - 1; j++) {
+            if (arr[j] <= pivot) {
+                i++;
+                int temp = arr[i];
+                arr[i] = arr[j];
+                arr[j] = temp;
+            }
+        }
+        int temp = arr[i + 1];
+        arr[i + 1] = arr[h];
+        arr[h] = temp;
+        return i + 1;
+
+    }
+
+    // Sorts arr[l..h] using iterative QuickSort
+    void QuickSort(int arr[], int l, int h) {
+        // Try using Stack Data Structure to remove recursion.

```

Comment:
Use stack here To implement Iterative Quick sort.

Reply:
Sir, should I redo and submit it?

Reply:
Just try to implement Quicksort without recursio If you can.